lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -326,6 +359,84 @@ static bool sinkCode(Function *F) {
       }
     }
 
+    // Sink Transpose below ChannelShuffle (Reshape->Transpose->Reshape)
+    if (auto *RN1 = dyn_cast<ReshapeNode>(node)) {
+      auto *TR1 = dyn_cast<TransposeNode>(RN1->getInput());
+      if (!TR1) {
+        continue;
+      }
+      auto *RN2 = dyn_cast<ReshapeNode>(TR1->getInput());
+      if (!RN2) {
+        continue;
+      }
+      auto *TR2 = dyn_cast<TransposeNode>(RN2->getInput());
+      if (!TR2) {
+        continue;
+      }
+
+      ChannelShuffleDimGroups rn2DimGroups;
+      if (!groupDimentionsFromChannelShuffleReshape(*RN2, rn2DimGroups)) {
+        continue;
+      }
+
+      // create newRN2
```

Comment:
Can you add more descriptive comments about what exactly is going on in each for loop here?